isis/tests/Fixtures.cpp

```diff
@@ -0,0 +1,38 @@
+#include "Fixtures.h"
+
+namespace Isis {
+  
+  void TestCube::SetUp() {
+    tempFile.open();
+    testCube.setPixelType(UnsignedByte);
+  }
+
+  void TestCube::TearDown() {
+    if (testCube.isOpen()) {
+      testCube.close();
+    }
+  }
+
+  void TestCube::createCube(Pvl &label) {
+    PvlObject cubeLabel = label.findObject("IsisCube");
+    PvlGroup dimensions = cubeLabel.findObject("Core").findGroup("Dimensions");
+    testCube.setDimensions(dimensions["Samples"],
+                          dimensions["Lines"],
+                          dimensions["Bands"]);
+    testCube.create(tempFile.fileName());
+
+    for (auto grpIt = cubeLabel.beginGroup(); grpIt!= cubeLabel.endGroup(); grpIt++) {
+      testCube.putGroup(*grpIt);
+    }
+  }
+
+  void TestCube::createCube(Pvl &label, nlohmann::json &isd) {
+    createCube(label);
+    testCube.attachSpiceFromIsd(isd);
+    
+    // close cube to flush and write tables
+    testCube.close();
+    testCube.open(tempFile.fileName()+".cub");
```

Comment:
We might run into more issues like this where we need to close the cubes and re-open after changes. Seems cubes were not designed to live prolonged lives as in-memory objects.